components/script/dom/cssstyledeclaration.rs

```diff
@@ -169,27 +218,23 @@ impl CSSStyleDeclaration {
                         declarations: declarations.into_iter().map(|d| (d, importance)).collect(),
                         important_count: important_count,
                     };
-                    self.owner.set_style_attr(block.to_css_string());
-                    *option = Some(Arc::new(RwLock::new(block)));
+                    if let CSSStyleOwner::Element(ref el) = self.owner {
+                        el.set_style_attr(block.to_css_string());
+                        *el.style_attribute().borrow_mut() = Some(Arc::new(RwLock::new(block)));
+                    }
```

Comment:
Put an `else` with a panic it it. There's an invariant here that only element owners can have empty PDBs, if that invariant changes in the future we should know about it.